TensorKitSectors/src/auxiliary.jl

```diff
@@ -13,3 +13,78 @@ function _kron(A, B)
     end
     return C
 end
+
+# Manhattan based distance enumeration: I is supposed to be one-based index
+# TODO: is there any way to make this faster?
```

Comment:
There's definitely some number theory stuff that could come in handy here, if I am not mistaken, the relevant terms are triangular numbers, tetrahedral numbers, and their generalizations.
There are general formulas for this:
```math
 \sum_{n_{k-1} = 1}^{n_k} \sum_{n_{k-2} = 1}^{n_{k-1}} \cdots \sum_{n_1 = 1}^{n_2} n_1 = \begin{pmatrix} n_k + k - 1 \\ k \end{pmatrix}
```

This is however for infinite grids, but I feel like it should be possible to split the grid into regions that are either hyperrectangular, or "hypertriangular", both of which have "volume formulas"?

https://en.wikipedia.org/wiki/Tetrahedral_number

Reply:
The binomial formula is what I started with, but that indeed only works if there are no upper bounds, i.e. if all the sectors involved are infinite.